### PR TITLE
ENH: provide a charter for DANDI

### PR DESCRIPTION
**Pull request by satra**
Status: WIP

cc: @yarikoptic

### CODE REVIEW

File: DANDI.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Objectives
+- Develop a cloud platform for versioned neurophysiology data storage for the purposes of collaboration, archiving, and preservation. 
+- Provide easy to use tools for neurophysiology data submission and access in the archive; and 
+- Facilitate adoption of NWB via standardized applications for data ingestion, visualization and processing.
+
+As an exercise, let’s assume you lose all the data in your lab. What would you want from the archive? Our hope is that your answer to this question, the necessary data and metadata that you need, is at least what we should be storing.
+
+Our goals for the archive, we think, are rather simple:
+
+- Optimize storage based on cost to produce and process.
```
Comment (tgbugs) [inline DANDI.md]:
```suggestion
- Superoptimize storage based on cost to produce and process.
```

File: DANDI.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Description
+DANDI is a platform for publishing, sharing, and processing neurophysiology data funded by the BRAIN Initiative. We will work with BICCN and other BRAIN awardees to curate data using community standards, and to make data FAIR (Findable, Accessible, Interoperable, and Reusable). We will use DANDI to help improve our understanding of the brain.
+
+# Roles
+- [Principal Investigator] Satrajit Ghosh (satra@mit.edu)
+- [Principal Investigator] Yaroslav Halchenko (yoh@dartmouth.edu)
+- [NWB and community liaison] Ben Dichter (ben.dichter@gmail.com)
+- [Co-Investigator] Michael Grauer (michael.grauer@kitware.com)
+
+## Definitions
```
Comment (satra) [inline DANDI.md:12]: - add BIDS, NIDM
- add INCF BIDS endorsement link
- add URLs to these words

File: DANDI.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+HDF - Hierarchical Data Format supports an unlimited variety of datatypes, and is designed for flexible, efficient I/O and for high volume and complex data.
+
+NWB - Neurodata Without Borders. The NWB file specification can capture the entirety of heterogeneous data representing an experiment and is built on top of HDF. 
+
+NWB:N - NWB:Neurophysiology, is a common file format that accommodates a wide variety of neurophysiology data as well as complex metadata related to stimulus and behavior. This is being developed under BRAIN award [1R24MH116922-01](https://projectreporter.nih.gov/project_info_description.cfm?aid=9582696).
+
+## Objectives
+- Develop a cloud platform for versioned neurophysiology data storage for the purposes of collaboration, archiving, and preservation.
```
Comment (satra) [inline DANDI.md]:
```suggestion
- Develop a cloud platform for versioned neurophysiology data storage for the purposes of collaboration and dissemination of data and software containers. 
```

File: DANDI.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+NWB:N - NWB:Neurophysiology, is a common file format that accommodates a wide variety of neurophysiology data as well as complex metadata related to stimulus and behavior. This is being developed under BRAIN award [1R24MH116922-01](https://projectreporter.nih.gov/project_info_description.cfm?aid=9582696).
+
+## Objectives
+- Develop a cloud platform for versioned neurophysiology data storage for the purposes of collaboration, archiving, and preservation. 
+- Provide easy to use tools for neurophysiology data submission and access in the archive; and 
+- Facilitate adoption of NWB via standardized applications for data ingestion, visualization and processing.
+
+As an exercise, let’s assume you lose all the data in your lab. What would you want from the archive? Our hope is that your answer to this question, the necessary data and metadata that you need, is at least what we should be storing.
+
+Our goals for the archive, we think, are rather simple:
```
Comment (TimothyTickle) [inline DANDI.md]: I am excited to learn about your aspirations and be a partner in making them happen but I feel the charter should focus on what work we are committed and not committed to doing.

File: DANDI.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# DANDI: Distributed Archives for Neurophysiology Data Integration
+
+## Description
+DANDI is a platform for publishing, sharing, and processing neurophysiology data funded by the BRAIN Initiative. We will work with BICCN and other BRAIN awardees to curate data using community standards, and to make data FAIR (Findable, Accessible, Interoperable, and Reusable). We will use DANDI to help improve our understanding of the brain.
```
Comment (TimothyTickle) [inline DANDI.md]: Could you call out specific kinds of data you are interested in working with?

File: DANDI.md (hunk 4 above)
Comment (satra) [inline DANDI.md]: - remove this section
-

File: DANDI.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# DANDI: Distributed Archives for Neurophysiology Data Integration
```
Comment (tgbugs) [inline DANDI.md:1]: @tgbugs provide list of data modalities to @satra

File: DANDI.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+- Develop infrastructure for ingesting, storing, searching across, and retrieving datasets containing cellular neurophysiology data and common data derivatives
+- Develop web portal to support data and workflow management
+- Support for BRAIN standards (NWB:N, BIDS, NIDM)
+- Data availability with at least file level atomicity
+- Provide human and programmatic search of data and metadata
+- Store derive data if it is of use to the community beyond a lab
+- Provide software to capture experimental metadata, convert to NWB, and upload/download data to/from the archive
+- Support browser-based visualization of data
+- Provide tutorials and training for users to use the archive
+- We will allow limited storage of private data with an embargo period, after which data becomes public
```
Comment (satra) [inline DANDI.md:43]: - when MRI or optical imaging enters this archive.

File: DANDI.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+## Out-of-scope
+- Develop neurophysiology standards
+- Develop neurophysiology software
+- Provide unlimited computing resources
```
Comment (satra) [inline DANDI.md:49]: - pure MRI datasets without any cellular level data

File: DANDI.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+- If users have to come back to the raw data, for example, to apply different processing strategies, etc., it should be feasible without going to the data producer.
+- The archive is not just an endpoint to dump data, it is intended as a living repository that enables collaboration within and across labs, and for others, the entry point for research.
+
+## In-scope
+- Develop infrastructure for ingesting, storing, searching across, and retrieving datasets containing cellular neurophysiology data and common data derivatives
+- Develop web portal to support data and workflow management
+- Support for BRAIN standards (NWB:N, BIDS, NIDM)
+- Data availability with at least file level atomicity
+- Provide human and programmatic search of data and metadata
+- Store derive data if it is of use to the community beyond a lab
```
Comment (TimothyTickle) [inline DANDI.md]: Please define derived data in definitions.

Comment (TimothyTickle) [inline DANDI.md]:
```suggestion
- Store derived data if it is of use to the community beyond a lab
```

File: DANDI.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+- Optimize storage based on cost to produce and process.
+- Enable reproducible practices and publications.
+- Enable secondary uses of data outside the intent of the study.
+- Reduce the need to contact data producers by enriching the data with rich metadata.
+- If users have to come back to the raw data, for example, to apply different processing strategies, etc., it should be feasible without going to the data producer.
+- The archive is not just an endpoint to dump data, it is intended as a living repository that enables collaboration within and across labs, and for others, the entry point for research.
+
+## In-scope
+- Develop infrastructure for ingesting, storing, searching across, and retrieving datasets containing cellular neurophysiology data and common data derivatives
```
Comment (TimothyTickle) [inline DANDI.md:34]: Is interactivity to the BCDC portal in scope?

File: DANDI.md (hunk 7 above)
Comment (TimothyTickle) [inline DANDI.md:43]: maybe change this to support the BICCN embargo policy (as the consortium defines it). That feels like a policy we will have listed here separately.

File: DANDI.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+As an exercise, let’s assume you lose all the data in your lab. What would you want from the archive? Our hope is that your answer to this question, the necessary data and metadata that you need, is at least what we should be storing.
+
+Our goals for the archive, we think, are rather simple:
+
+- Optimize storage based on cost to produce and process.
+- Enable reproducible practices and publications.
+- Enable secondary uses of data outside the intent of the study.
+- Reduce the need to contact data producers by enriching the data with rich metadata.
+- If users have to come back to the raw data, for example, to apply different processing strategies, etc., it should be feasible without going to the data producer.
+- The archive is not just an endpoint to dump data, it is intended as a living repository that enables collaboration within and across labs, and for others, the entry point for research.
```
Comment (satra) [inline DANDI.md]: + metadata/data exchange with other archives, compute services, and coordinating centers

**Review by carolth — COMMENTED** [review]
Suggestions: 
1) Add definitions for NIDM standards and BIDS
2) Would like to request that in scope is supporting data exchange with BCDC and or other archives as needed to support linking patchseq data as well as to enable BCDC integration to enable a user at Cell Registry to be directed to a subset of data at DANDI

File: DANDI.md (hunk 7 above)
Comment (satra) [reply DANDI.md:43]: i would like to discuss this today if possible. i am not sure this has been defined yet.

File: DANDI.md (hunk 6 above)
Comment (satra) [reply DANDI.md:1]: @tgbugs - let me know if there is a specific way to encode this.

File: DANDI.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+## Objectives
+- Develop a cloud platform for versioned neurophysiology data storage for the purposes of collaboration and dissemination of data and software containers. 
+- Provide easy to use tools for neurophysiology data submission and access in the archive; and 
+- Facilitate adoption of NWB via standardized applications for data ingestion, visualization and processing.
+
+As an exercise, let’s assume you lose all the data in your lab. What would you want from the archive? Our hope is that your answer to this question, the necessary data and metadata that you need, is at least what we should be storing.
+
+## In-scope
+- Data modalities: multi-channel or single channel timeseries and 3D/4D imaging data
```
Comment (satra) [inline DANDI.md:33]: @tgbugs - let me know how you would want this encoded. i think it should go in a predefined separate section.

File: DANDI.md (hunk 10 above)
Comment (satra) [reply DANDI.md:34]: added on in response to this question and @carolth comments.

File: DANDI.md (hunk 7 above)
Comment (TimothyTickle) [reply DANDI.md:43]: I added developing an Embargo Policy as an issue to the repo.

File: DANDI.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+Derived data. Derived data are files/objects created by software transformations of the data captured by a data capturing device (electrode, microscope, MRI scanner) or other derived data. 
+
+## Objectives
+- Develop a cloud platform for versioned neurophysiology data storage for the purposes of collaboration and dissemination of data and software containers. 
+- Provide easy to use tools for neurophysiology data submission and access in the archive; and 
+- Facilitate adoption of NWB via standardized applications for data ingestion, visualization and processing.
+
+As an exercise, let’s assume you lose all the data in your lab. What would you want from the archive? Our hope is that your answer to this question, the necessary data and metadata that you need, is at least what we should be storing.
+
+## In-scope
```
Comment (lydiang) [inline DANDI.md:32]: For consideration

Support API to access indexed timeseries traces from select dataset to enable the visualization and analysis of timeseries in third party system such as the BCDC Cell Registry

Comment (satra) [reply DANDI.md:32]: @lydiang - we will likely support this, but the api development will take a bit of time. @mgrauer, @yarikoptic @bendichter and i will be having meetings with people during SFN to guide the development of the API,  so this kind of feedback would be much appreciated.

**Pull request merged**